### ISSUE TITLE
2.8.x / 2.9.x release?

### ISSUE THREAD

**Issue by SinisterRectus** [luvit/luvi#187]
I just noticed that there is a tag for 2.8.0, but no release, and therefore no luvit release that contains luvi 2.8.0. Is this intentional?

**Comment by rphillips** [comment]
Not intentional. The build machines have been gone for some time. It would be nice to get travis to push images.

**Comment by rphillips** [comment]
@SinisterRectus which platform are you interested in?

**Comment by aiverson** [comment]
I have a decent amount of experience in compiling and writing C software. I can potentially help out here.

**Comment by rphillips** [comment]
Travis just added beta support for windows... We could potentially leverage that.

**Comment by joerg-krause** [comment]
There are already 46 commits to master since tag 2.8.0. Maybe it is time for 2.9.0?

**Comment by SinisterRectus** [comment]
2.8.1 or 2.9.0 might be appropriate, depending on what the changes are. Would probably be best to have a reliable build procedure for building 2.8.0 and then worry about the rest.

**Comment by joerg-krause** [comment]
We are stuck to version 2.7.6 in Buildroot, as it is the latest version which provides a release tarball. Unfortunately, this version does not compile with OpenSSL 1.1.x

I'd be delighted if there was a new release with a downloadable tarball :pray:

**Comment by SinisterRectus** [comment]
[This](https://github.com/luvit/luvi/releases/tag/v2.8.0)?

**Comment by joerg-krause** [comment]
No, as the deps folders are all empty.

**Comment by joerg-krause** [comment]
[This](https://github.com/luvit/luvi/releases/download/v2.7.6/luvi-src-v2.7.6.tar.gz).

**Comment by joerg-krause** [comment]
The Makefile already has the necessary target `publish-src` to upload a release tarball containing all the source files.

Maybe one of the maintainers can tag a release and run `make publish-src`, so can get an updated version of luvi, which can be build with OpenSSL 1.1.1a.

**Comment by rphillips** [comment]
2.8.1 or 2.9.0?

**Comment by joerg-krause** [comment]
I tend towards 2.9.0.

**Comment by joerg-krause** [comment]
After merging #189

**Comment by creationix** [comment]
Let's get this out.  At a minimum we can publish the tarball for builtroot.

The binaries are harder depending on how many we want to support.  I had a pile of raspberry pi machines, 64-bit arm boards and virtualbox images that I was using before to manually build all the variants.

**Comment by creationix** [comment]
@joerg-krause is anything extra needed to support the newer openssl or is it already in the newer luvi commits?

**Comment by joerg-krause** [comment]
@creationix No, I think everything is prepared. Buildroot uses an external OpenSSL, but still requires the bundled lua-openssl, which has been bumped to the latest version.

Thanks for taking care!

**Comment by creationix** [comment]
@joerg-krause please test https://github.com/luvit/luvi/releases/tag/v2.9.0

It's been a while and my old scripts/tools aren't working.  I created this release manually and uploaded the tarball manually, but I think it will work.

**Comment by creationix** [comment]
I also manually built and uploaded linux x64 binaries on an ubuntu 14.04 system.  The docker-based stuff isn't working for me.

**Comment by joerg-krause** [comment]
@creationix Thanks for the source tarball! It is working perfectly, but the `VERSION` is not set to `2.9.0` in the tarball, but to `v2.7.6-61-gfe02ab2`:

```
# luvi --version
luvi v2.7.6-61-gfe02ab2
zlib: 1.2.11
libuv: 1.25.0
ssl: OpenSSL 1.1.1a  20 Nov 2018, lua-openssl 0.7.3
```

**Comment by creationix** [comment]
Ok, rebuilt the tarball.  I think I had built it before creating the new tag in git.

**Comment by joerg-krause** [comment]
Thanks! That worked:

```
# luvi --version
luvi v2.9.0
zlib: 1.2.11
libuv: 1.25.0
ssl: OpenSSL 1.1.1a  20 Nov 2018, lua-openssl 0.7.3
```

**Comment by squeek502** [comment]
@creationix here's a full set of Windows binaries/libs if you want to add them to the release: https://www.ryanliptak.com/misc/luvi-windows-2.9.0.zip

Built with MSVC 2015 Update 3 on Windows 7 and includes [this fix](https://github.com/luvit/luvi/pull/191).

```
> luvi --version
luvi v2.9.0
libuv: 1.25.0
rex: 8.37 2015-04-28
winsvc: 1.0.0
ssl: OpenSSL 1.1.0i  14 Aug 2018, lua-openssl 0.7.3
```

**Comment by squeek502** [comment]
Here are Windows binaries/libs for 2.9.1 (the same as the 2.9.0 ones above except for the version string):
https://www.ryanliptak.com/misc/luvi-windows-2.9.1.zip

```
> luvi --version
luvi v2.9.1
libuv: 1.25.0
rex: 8.37 2015-04-28
winsvc: 1.0.0
ssl: OpenSSL 1.1.0i  14 Aug 2018, lua-openssl 0.7.3
```

**Comment by creationix** [comment]
I think my linux binaries are wrong.  They appear to have dynamically linked libuv.

**Comment by squeek502** [comment]
> I think my linux binaries are wrong. They appear to have dynamically linked libuv.

Yeah, they would have since you built them before https://github.com/luvit/luvi/pull/191. @rphillips released v2.9.1 with that fix.

**Comment by creationix** [comment]
Thanks, I published linux binaries for the v2.9.1 release.  They also appear to dynamically link openssl, but I think that's on purpose now for linux?

```
~/Code/luvi$ ldd build/luvi 
	linux-vdso.so.1 =>  (0x00007ffdb10aa000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fa161f53000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fa161d35000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fa161b31000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fa16182b000)
	libssl.so.1.0.0 => /lib/x86_64-linux-gnu/libssl.so.1.0.0 (0x00007fa1615cc000)
	libcrypto.so.1.0.0 => /lib/x86_64-linux-gnu/libcrypto.so.1.0.0 (0x00007fa1611ef000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fa160fd7000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fa160c0e000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fa16215b000)
```

**Comment by squeek502** [comment]
> Thanks, I published linux binaries for the v2.9.1 release. They also appear to dynamically link openssl, but I think that's on purpose now for linux?

[That's the default now, yeah](https://github.com/luvit/luvi/issues/178). You can add these Windows binaries to the release too if you'd like: https://www.ryanliptak.com/misc/luvi-windows-2.9.1.zip

**Comment by rphillips** [comment]
Thanks! Just for sanity, I ran the binaries through clamav.
